components/admin_console/system_user_detail/index.ts

```diff
@@ -2,43 +2,43 @@
 // See LICENSE.txt for license information.
 
 import {connect} from 'react-redux';
-import {bindActionCreators} from 'redux';
+import {bindActionCreators, Dispatch} from 'redux';
 
-import {getUser, getProfiles} from 'mattermost-redux/selectors/entities/users';
+import {getUser} from 'mattermost-redux/selectors/entities/users';
 import {updateUserActive} from 'mattermost-redux/actions/users';
 import {getConfig} from 'mattermost-redux/selectors/entities/general';
 import {addUserToTeam} from 'mattermost-redux/actions/teams';
 
-import {setNavigationBlocked, deferNavigation, cancelNavigation, confirmNavigation} from 'actions/admin_actions.jsx';
-import {getNavigationBlocked, showNavigationPrompt} from 'selectors/views/admin';
+import {GenericAction} from 'mattermost-redux/types/actions';
 
-import SystemUserDetail from './system_user_detail.jsx';
+import {GlobalState} from 'mattermost-redux/types/store';
 
-function mapStateToProps(state, ownProps) {
+import {setNavigationBlocked} from 'actions/admin_actions.jsx';
+
+import SystemUserDetail from './system_user_detail';
+
+type OwnProps = {
+    match: any
+};
+
+function mapStateToProps(state: GlobalState, ownProps: OwnProps) {
     const config = getConfig(state);
     const userId = ownProps.match.params.user_id;
     const user = getUser(state, userId);
     return {
         user,
         mfaEnabled: config.EnableMultifactorAuthentication === 'true',
-        navigationBlocked: getNavigationBlocked(state),
-        showNavigationPrompt: showNavigationPrompt(state),
     };
 }
 
-function mapDispatchToProps(dispatch) {
+function mapDispatchToProps(dispatch: Dispatch<GenericAction>) {
     return {
         actions: bindActionCreators({
-            getUser,
-            getProfiles,
             updateUserActive,
             setNavigationBlocked,
-            deferNavigation,
-            cancelNavigation,
-            confirmNavigation,
             addUserToTeam,
         }, dispatch),
-    };
+    } as any;
```

Comment:
This still shouldn't require an any. If you are getting an error here you may have to use the `ActionCreatorsMapObject` and manually define the types for the actions specified. Have a look at this file for an example: https://github.com/mattermost/mattermost-webapp/blob/master/components/sidebar/index.ts

Reply:
I removed the any, but my solution is not elegant.
`setNavigationBlocked` is typed differently from `updateUserActive` and `addUserToTeam` and this results in the `ActionCreatorsMapObject` not being able to return correct types.
I binned these groups of actions separately and then assigned them to the props.